src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -24,7 +24,10 @@ public class TagHelperRunner
                 executionContext.AllAttributes,
                 executionContext.UniqueId,
                 executionContext.GetChildContentAsync);
-            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes);
+            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes)
+            {
+                SelfClosing = executionContext.SelfClosing,
```

Comment:
I understand that. I really don't feel like we should though. `SelfClosing` is a very unimportant property as far as `TagHelperOutput` is concerned. If anything it's just additive. My reasoning in one way falls in line with what we have in MVC for `TagBuilder`, the default `ToString()` doesn't require any mention of rendering the tag self closing or not (`TagRenderMode.Normal`). We follow the same approach here were it's defaulted to not being self-closing and you can change later if you want.